Yc 384

nicolas-sitylb opened the pull request:
Fix # 384 in Jira.

monodo commented on permits/admin.py:
```diff
@@ -767,6 +840,14 @@ def __new__(cls, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
+        super(PermitAdministrativeEntityAdminForm, self).__init__(*args, **kwargs)
+        self.fields["expeditor_name"].disabled = False
+        self.fields["expeditor_name"].initial = None
```
useful ? if not, please remove

nicolas-sitylb replied:
Done.

monodo commented on permits/admin.py:
```diff
@@ -223,6 +228,17 @@ def get_queryset(self, request):
+            obj.is_staff = False
+            for group in form.cleaned_data["groups"]:
+                if group.permitdepartment.is_integrator_admin:
+                    obj.is_staff = True
```
If `obj.is_superadmin`, do nothing. Otherwise, superadmin could loose it's admin access

monodo replied:
https://docs.djangoproject.com/en/4.0/ref/contrib/auth/#django.contrib.auth.models.User.is_superuser

nicolas-sitylb replied:
Resolved by https://github.com/yverdon/geocity/pull/383/commits/7a2f14fd1dac30ea60bd329856740907910bfe15

monodo commented on permits/admin.py:
```diff
@@ -364,11 +380,45 @@ def clean_permissions(self):
+        "get__integrator",
+        "get__mandatory_2fa",
+    ]
+
```
If not too complicated, change the display to checkbox like in WOTs for instance (1.4)

![image](https://user-images.githubusercontent.com/3356536/154935597-2219bf00-ab24-4526-a525-06ba4fae0b8d.png)

monodo reviewed (changes requested):
2 little points to check/fix

nicolas-sitylb commented:
Fixed `is_staff` for super users + synchronized with the `main` branch.    
(also added an `order` field in the WorksObject model which could serve as a basis to order the works objects).
